CMakeLists.txt

```diff
@@ -59,7 +58,7 @@ set(WPEBACKEND_FDO_LIBRARIES
     ${GLIB_LIBRARIES}
     ${WAYLAND_LIBRARIES}
     ${WAYLAND_EGL_LIBRARIES}
-    ${WPE_LIBRARIES}
+    PkgConfig::WPE
```

Comment:
As discussed on IRC, this naming is lame. Don suggested creating an alias in FindWPE.cmake so nothing outside the find module needs to write PkgConfig. Perhaps WPE::libwpe.

Reply:
I'll be updating this later to use `WPE::libwpe` :ok_hand:

Reply:
Well, I have been trying the following creates an alias:

```cmake
add_library(WPE::libwpe ALIAS PkgConfig::PC_WPE)
```

…but then `PkgConfig::PC_WPE` **needs** be be a global target. With CMake ≥3.13 we can pass `GLOBAL` to `pkg_check_modules()` but that won't work with older versions, and AFAIU we want to support CMake 3.10.x because that's what our older supported distributions ship.

It will take me a little longer, but it seems for compatibility with older versions I am going to need manually defining a new imported library target (not an alias), and then copy the properties from `PkgConfig::PC_WPE` to it.

Reply:
> It will take me a little longer, but it seems for compatibility with older versions I am going to need manually defining a new imported library target (not an alias), and then copy the properties from `PkgConfig::PC_WPE` to it.

Well it would be silly to do this much manual work for every find module in WebKit. The goal here is to use this as a prototype for restructuring all of our find modules, so I think we should talk with Don before resorting to such a workaround.

But my interpretation of our findings thus far is "lol find modules have never worked properly and there is no idiomatic or canonical way to write them," so I guess I should lower my expectations.

Reply:
This was dramatically easier than expected in the end. Basically, when the library is found and the `WPE::libwpe` imported library target needs to be defined, boils down to: 

```cmake
add_library(WPE::libwpe INTERFACE IMPORTED)
if (TARGET PkgConfig::WPE)  # Defined by pkg_check_modules, if module found.
    target_link_libraries(WPE::libwpe INTERFACE PkgConfig::WPE)
else ()  # Use the paths from find_path/find_library
    set_property(TARGET WPE::libwpe PROPERTY
        INTERFACE_LINK_LIBRARIES "${WPE_LIBRARY}")
    set_property(TARGET WPE::libwpe PROPERTY
       INTERFACE_INCLUDE_DIRECTORIES "${WPE_INCLUDE_DIR}")
endif ()
```

This avoids needing `ALIAS` at all, because dependencies are transitive: using `WPE::libwpe` transitively adds the flags from `PkgConfig::WPE` as needed.

As a bonus, with this approach the `Find<xyz>.cmake` can add and modify properties to `WPE::libwpe` on top of what `pkg_check_modules()` has already configured for `PkgConfig::WPE`—right now I cannot think of an use-case, but the possibility is there.

Reply:
But doesn't it discard CFLAGS_OTHER and LDFLAGS_OTHER? It's just not enough.

Reply:
The `PkgConfig::WPE` imported target carries all the flags (including `_CFLAGS_OTHER` and `_LDFLAGS_OTHER`) if I am reading the `FindPkgConfig.cmake` file included with CMake correctly. That means that “depending on” `WPE::libwpe` will bring the flags to the compiler and linker command lines, because it in turn depends on `PkgConfig::WPE`.

In the fallback case where `pkg_check_modules()` failed, but `find_path()` and `find_library()` succeeded, we only know where the library and the headers are so that is why only the `INTERFACE_LINK_LIBRARIES` and `INTERFACE_INCLUDE_DIRECTORIES` are set—we do not know anything else.